### DISCUSSION TITLE
ScratchAddons for mobile?

### DISCUSSION THREAD

**Discussion by jemunm378** [ScratchAddons/ScratchAddons#343] [Help]
My question is can I fork the "ScratchAddons website" repository to create my own one for mobile users? I was thinking of using IOS Shortcuts to make the Addons.

**Comment by retronbv** [comment]
how would that work

**Comment by jemunm378** [comment]
> how would that work

Well, I was thinking of using IOS shortcuts app to use it. It’s kind of like chrome extensions, but for IOS

**Comment by GrahamSH-LLK** [comment]
If you can make that work, that would be really cool! Since this is open source, anyone can fork it. I _think_ you may be able to use this extension on firefox on mobile, but I'm not sure.

**Comment by infinitytec** [comment]
It should work on Firefox for Android. There is some sort of validation or approval process though.

**Comment by WorldLanguages** [comment]
I believe Firefox for Android is the way to go.
Extension support for iOS and iPadOS appears to be non-existent, and it's also prohibited because of Apple's guidelines.
Safari 14 for MacOS does support WebExtensions, however it's too early to have a say on that. Some important APIs used by Scratch Addons aren't implemented on Safari yet, hopefully in the next version? Also, I hope they won't be requiring $99 a year to publish extensions into the store. Hopefully there's some non painful way to sideload them?

**Comment by GrahamSH-LLK** [comment]
@WorldLanguages Yeah, no. There's literally no way Apple is letting you do that. This is the company that won't let you use "unsigned" apps. Unless there's some kind of opensource/student deal, I don't think this is a reality. (I mean, there could be _some_ way to sideload, but I'm sure it would be painful.

**Comment by retronbv** [comment]
Shortcuts don't need to be signed or side loaded

**Comment by WorldLanguages** [comment]
I don't think shortcuts and brower extensions are somehow related anyway?

**Comment by retronbv** [comment]
You can’t make a shortcut that automatically changes things, you need to run the shortcut for it to modify/add things.

**Comment by GrahamSH-LLK** [comment]
I don't think Shortcuts have _any_ control over the CSS/JS/HTML of pages, although I could be wrong. If there is any support, it'll be limited.

**Comment by GrahamSH-LLK** [comment]
Never mind! I guess you could replicate it _somewhat_ with https://support.apple.com/guide/shortcuts/use-the-run-javascript-on-webpage-action-apdb71a01d93/ios. Like I said, it would be super limited.

**Comment by WorldLanguages** [comment]
Yeah, Scratch Addons won't work without a background page - I doubt Shortcuts supports that.
